doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -144,61 +144,142 @@ presentation of Cic is given in Chapter :ref:`calculusofinductiveconstructions`.
-   cofix_body       : `ident` [`binders`] [: `term`] := `term`
-   annotation       : { struct `ident` }
-   match_item       : `term` [as `name`] [in `qualid` [`pattern` … `pattern`]]
-   dep_ret_type     : [as `name`] `return_type`
-   return_type      : return `term`
-   equation         : `mult_pattern` | … | `mult_pattern` => `term`
-   mult_pattern     : `pattern` , … , `pattern`
-   pattern          : `qualid` `pattern` … `pattern`
-                    : @ `qualid` `pattern` … `pattern`
-                    : `pattern` as `ident`
-                    : `pattern` % `ident`
-                    : `qualid`
-                    : _
-                    : `num`
-                    : ( `pattern` | … | `pattern` )
+   term : forall `open_binders` , `term`
+        : fun `open_binders` => `term`
+        : let `name` `binder_list_opt` `type_cstr` := `term` in `term`
+        : let `single_fix` in `term`
+        : let `name_alt` `return_type_opt` := `term` in `term`
+        : let ' `pattern` := `term` in `term`
+        : let ' `pattern` := `term` return `term100` in `term`
+        : let ' `pattern` in `pattern` := `term` return `term100` in `term`
+        : if `term` `return_type_opt` then `term` else `term`
+        : `single_fix`
+        : `single_fix` with `fix_body_list` for `ident`
+        : `term100`
+   type_cstr : : `term`
+             : `empty`
+   name : _
+        : `ident`
+   name_alt : ( `name_list_comma_opt` )
+            : ()
+   name_list_comma_opt : `name_list_comma`
+                       : `empty`
+   name_list_comma : `name_list_comma` , `name`
+                   : `name`
+   open_binders : `name` `name_list_opt` : `term`
+                : `name` `name_list_opt` `binder_list_opt`
+                : `name` .. `name`
+                : `closed_binder` `binder_list_opt`
+   name_list_opt : `name_list_opt` `name`
+                 : `empty`
+   binder_list_opt : `binder_list_opt` `binder`
+                   : `empty`
+   binder : `name`
+          : `closed_binder`
+   closed_binder : ( `name` `name_list` : `term` )
+                 : ( `name` : `term` )
+                 : ( `name` := `term` )
+                 : ( `name` : `term` := `term` )
+                 : { `name` }
+                 : { `name` `name_list` : `term` }
+                 : { `name` : `term` }
+                 : { `name` `name_list` }
+                 : `( `typeclass_constraint_list_comma` )
+                 : `{ `typeclass_constraint_list_comma` }
+                 : ' `pattern0`
+                 : ( _ : `term` | `term` )
+   return_type_opt : `as_name_opt` return `term100`
+                   : `empty`
+   as_name_opt : as `name`
+               : `empty`
+   single_fix : `fix_kw` `fix_body`
+   fix_body_list : `fix_body_list` with `fix_body`
+                 : `fix_body`
+   fix_kw : fix
+          : cofix
+   fix_body : `ident` `binders_fixannot` `type_cstr` := `term`
+   name_list : `name_list` `name`
+             : `name`
```

Comment:
Related to the binder chunk.

Reply:
But also used in

```
   impl_ident_tail : }
                   : `name_list` : `term` }
                   : `name_list` }
                   : : `term` }
```

If you want to repeat `name_list` both under term and binder_constr/term_let, productionlist requires distinct names.